### ISSUE TITLE
docker load not working

### ISSUE THREAD

**Issue by thunderbird85** [Microsoft/Virtualization-Documentation#355]
Hi,
I have try the description from 
Virtualization-Documentation/virtualization/windowscontainers/quick_start/quick_start_windows_10.md
but there is a problem on Step 3 
`docker load -i nanoserver.tar.gz`

I got the following error
```
PS C:\WINDOWS\temp> docker load -i .\nanoserver.tar.gz
58684737b3d1: Loading layer [==================================================>] 1.007 GB
re-exec error: exit status 1: output: ProcessUtilityVMImage C:\ProgramData\docker\windowsfilter\5d73fa815a57e4506ad2457c
fd875f7b6bd2c0dee7e0050beadc754566bb63d0\UtilityVM: The process cannot access the file because it is being used by another process.
```
If I try it with docker pull it's the same:

```
PS C:\WINDOWS\temp> docker pull microsoft/nanoserver
Using default tag: latest
latest: Pulling from microsoft/nanoserver

cf62dbf6d334: Extracting [==================================================>] 358.4 MB/358.4 MB
failed to register layer: re-exec error: exit status 1: output: ProcessUtilityVMImage C:\ProgramData\docker\windowsfilte
r\a5dfe6399ddf943c54bc1330562c7095c29b3bdadb74673f56a1257ac6b916bd\UtilityVM: The process cannot access the file because
 it is being used by another process.
```

The PowerShell ist startet as administrator.

Docker version:
```
PS C:\WINDOWS\system32> docker version
Client:
 Version:      1.12.0
 API version:  1.24
 Go version:   go1.6.3
 Git commit:   8eab29e
 Built:        Thu Jul 28 23:54:00 2016
 OS/Arch:      windows/amd64

Server:
 Version:      1.12.0
 API version:  1.24
 Go version:   go1.6.3
 Git commit:   8eab29e
 Built:        Thu Jul 28 23:54:00 2016
 OS/Arch:      windows/amd64
```

Windows Insider-Build 14393.67

**Comment by PatrickLang** [comment]
Do you have any other images installed? I haven't seen this problem since we released put `microsoft/nanoserver` on Docker Hub last week

`docker pull microsoft/nanoserver` is the right way to do this with Docker v1.12 on your build

@enderb-ms can you watch this issue?

**Comment by thunderbird85** [comment]
There is no other image installed. I habe try it with windowsservercore but it's the same
```
PS C:\WINDOWS\system32> docker pull microsoft/windowsservercore
Using default tag: latest
latest: Pulling from microsoft/windowsservercore
1239394e5a8a: Extracting [==================================================>] 3.461 GB/3.461 GB
failed to register layer: re-exec error: exit status 1: output: ProcessUtilityVMImage C:\ProgramData\docker\windowsfilte
r\aa05cc75c4fa4660b44e4407f55c0038fddad2a5d57d0cb7dc552f6076f63950\UtilityVM: The process cannot access the file because
 it is being used by another process.
```

**Comment by zuckerthoben** [comment]
I have the same problem

`PS C:\Windows\system32> docker pull microsoft/nanoserver
Using default tag: latest
latest: Pulling from microsoft/nanoserver

cf62dbf6d334: Extracting [==================================================>] 358.4 MB/358.4 MB
failed to register layer: re-exec error: exit status 1: output: ProcessUtilityVMImage C:\ProgramData\docker\windowsfilte
r\85b915e504d982a9df9ee6865995480369c5d1cdef651c3547c278607d56b59b\UtilityVM: The process cannot access the file because
 it is being used by another process.
PS C:\Windows\system32>`

**Comment by wangosu** [comment]
Any update or work around on this? I got the same error.

**Comment by StefanScherer** [comment]
@wangosu Have a look st the Windows 10 installation docs. It has been updated to download docker 1.13.0-dev zip file which fixes this issue.

**Comment by wangosu** [comment]
@StefanScherer  I did exactly as stated in https://msdn.microsoft.com/virtualization/windowscontainers/quick_start/quick_start_windows_10
with the 1.13.0-dev.zip
but still have the same error

**Comment by thunderbird85** [comment]
I have try it with 1.13.0 too 
and for me it's the same error

**Comment by StefanScherer** [comment]
Still struggling at `docker load`?
Or using the `docker pull microsoft/nanoserver` command?

I've just spun up an empty Win10 1607 VM and installed Docker and pulled the nanoserver image and it worked fine. 

Details of my setup can be found in this [Vagrant](https://github.com/StefanScherer/docker-windows-box/tree/master/windows10) environment and especially in this [`install-docker.ps1`](https://github.com/StefanScherer/docker-windows-box/blob/master/windows10/scripts/install-docker.ps1) script.

What does `docker info` and `docker images` show on your machine?

**Comment by wangosu** [comment]
My issue is docker pull, I tried both microsoft/nanoserver, microsoft/iis; after the image is extracted, both gave error:" failed to register layer; re-exec eror: exiy status 1: output: ProcessUtilityVMImage C:\ProgramData\docker\windowsfilter\...\UtilityVM: The process cannot access the file because
 it is being used by another process."

**Comment by VladM-L** [comment]
I am getting exactly the same error using both 1.12.0 and 1.13.0-dev. 
Interestingly, the folder and file it is complaining about are not there, i.e. the folder C:\ProgramData\Docker\windowsfilter is empty on my machine. 
I tried disabling antivirus, but it didn't change anything. 

I had Docker for Windows installed previously and it was working fine for Linus based images. But I need to run windows based apps, so I uninstalled that, installed Windows Containers and get this error for all windows images that I tried.

Please help. Thank you!

**Comment by wangosu** [comment]
Here is what i found after trying different ways:
1. On VM, works when pull images; but failed to run because VM does not have Hyper-V. So it works as expected.
2. On my laptop, upgraded to build 1607, still have the same issue when pull images no matter how many times I re-installed docker. 
3. On my surface Pro, upgraded to build 1607 also. Surprisingly everything works. 
I don't know why works on Surface but not on laptop. Both has the same OS build of 14393.105.

**Comment by VladM-L** [comment]
I found another thread about the same issue and someone there mentioned that it started to work after he completely removed antivirus software: https://social.msdn.microsoft.com/Forums/en-US/13523ccf-baf4-49fc-9e73-1df07e512304/reexec-error-when-trying-to-load-nanoserver-image?forum=windowscontainers
I.e. disabling Antivirus or using exceptions doesn't help, but removing it does. 
I am going to try it if I get green light from our IT.

**Comment by wangosu** [comment]
VladM-L, removed the antivirus did help. It is working for me now. Thanks

**Comment by VladM-L** [comment]
You are welcome wangosu!
It worked for me too after I uninstalled Symantec Endpoint Protection.

**Comment by msnelling** [comment]
Unfortunately for me I'm also having the same issue, but uninstalling antivirus an option. Wouldn't it be better to get to the bottom of the issue rather than asking people to compromise their security?

**Comment by antfurn** [comment]
Yep - I'm hitting this error too :-(

Fresh Win10 install but must run Symantec Endpoint Proctection (corporate policy).
So I'm stuffed until this is fixed!

PS C:\Program Files\Docker\Docker> docker version
Client:
 Version:      1.12.1
 API version:  1.24
 Go version:   go1.6.3
 Git commit:   23cf638
 Built:        Thu Aug 18 17:32:24 2016
 OS/Arch:      windows/amd64
 Experimental: true

Server:
 Version:      1.13.0-dev
 API version:  1.25
 Go version:   go1.7.1
 Git commit:   c2decbe
 Built:        Tue Sep 13 15:12:54 2016
 OS/Arch:      windows/amd64
PS C:\Program Files\Docker\Docker>

**Comment by aphophas** [comment]
I was able to get around this by just adding a folder exclusion for "C:\ProgramData\docker" in SEP on my box. Once I did that, the temp file being locked was released, and I was able to successfully pull, extract and view the image on my box. I hope this helps out, as I agree that fully uninstalled SEP or any AV product is not a valid workaround.

**Comment by msnelling** [comment]
@aphophas I tried the same thing and it didn't help in my case, only a full uninstall did the trick unfortunately.

**Comment by dguidara** [comment]
I can confirm that Unintalling Norton fixes it. Would like to be able to have norton installed though.

**Comment by tutnes** [comment]
I can confirm that temporarily disabling BitDefender mitigated the issue.

**Comment by rchhatwal** [comment]
Added folder exclusions for "C:\ProgramData\docker" & "C:\ProgramData\docker". Still getting the same error:
_failed to register layer: re-exec error: exit status 1: output: ProcessUtilityVMImage C:\ProgramData\docker\windowsfilte
r\f78c774278a7fd05393db8b6e9d3b66b74cf635cab5fcf1c1a570628ccf12a92\UtilityVM: The process cannot access the file because it is being used by another process.__

**Comment by RyanNewsom** [comment]
+1 for me. Tried disabling Norton, but of course is still had running processes. Had to fully uninstall it, then redid the pull and it worked.
Docker: 1.12.3

**Comment by dschilling** [comment]
I was having this same issue.  I have Symantec Endpoint Protection - and it's password protected so I can't uninstall or change it's exclusions.  However, the problem went away after I upgraded to Docker 1.14!  https://master.dockerproject.org/windows/amd64/docker-1.14.0-dev.zip

    PS C:\WINDOWS\system32> docker version
    Client:
     Version:      1.12.1
     API version:  1.24
     Go version:   go1.6.3
     Git commit:   23cf638
     Built:        Thu Aug 18 17:52:38 2016
     OS/Arch:      windows/amd64
    
    Server:
     Version:      1.14.0-dev
     API version:  1.26
     Go version:   go1.7.3
     Git commit:   e349d3d
     Built:        Mon Nov 21 19:30:47 2016
     OS/Arch:      windows/amd64

**Comment by jplonghi** [comment]
@dschilling do you have instructions or link on how to perform the upgrade?


```
PS C:\WINDOWS\system32> docker version
Client:
 Version:      1.13.0-rc2
 API version:  1.25
 Go version:   go1.7.3
 Git commit:   1f9b3ef
 Built:        Wed Nov 23 17:40:58 2016
 OS/Arch:      windows/amd64

Server:
 Version:             1.13.0-rc2
 API version:         1.25
 Minimum API version: 1.24
 Go version:          go1.7.3
 Git commit:          1f9b3ef
 Built:               Wed Nov 23 17:40:58 2016
 OS/Arch:             windows/amd64
 Experimental:        false
```

**Comment by jabbera** [comment]
@jplonghi if you figure that out let me know please.

**Comment by StefanScherer** [comment]
To download the latest zip file have a look at https://msdn.microsoft.com/en-us/virtualization/windowscontainers/docker/configure_docker_daemon#manual-installation

**Comment by philofinfinitejest** [comment]
I am getting the same behavior with version 1.14

**Version**

Client:
 Version:      1.14.0-dev
 API version:  1.26
 Go version:   go1.7.3
 Git commit:   e308f6b
 Built:        Tue Nov 22 18:15:55 2016
 OS/Arch:      windows/amd64

Server:
 Version:             1.14.0-dev
 API version:         1.26
 Minimum API version: 1.24
 Go version:          go1.7.3
 Git commit:          e308f6b
 Built:               Tue Nov 22 18:15:55 2016
 OS/Arch:             windows/amd64
 Experimental:        false


**Error**

[16:48:51.689][WindowsDockerDaemon][Info   ] time="2016-12-07T16:48:51.688867300-08:00" level=error msg="Error trying v2 registry: failed to register layer: re-exec error: exit status 1: output: ProcessUtilityVMImage C:\\ProgramData\\Docker\\windowsfilter\\4a3cff26991a967f76d0ae1eb8daf8115632bc769df418fb1eb21f756e07f131\\UtilityVM: The process cannot access the file because it is being used by another process." 
[16:48:51.689][WindowsDockerDaemon][Info   ] time="2016-12-07T16:48:51.688867300-08:00" level=error msg="Attempting next endpoint for pull after error: failed to register layer: re-exec error: exit status 1: output: ProcessUtilityVMImage C:\\ProgramData\\Docker\\windowsfilter\\4a3cff26991a967f76d0ae1eb8daf8115632bc769df418fb1eb21f756e07f131\\UtilityVM: The process cannot access the file because it is being used by another process."

**Comment by jabbera** [comment]
I had to uninstall SEP

**Comment by jplonghi** [comment]
Upgrade to 1.14.0-dev, but still the problem

```
PS C:\WINDOWS\system32> docker version
Client:
 Version:      1.14.0-dev
 API version:  1.26
 Go version:   go1.7.3
 Git commit:   e308f6b
 Built:        Tue Nov 22 18:15:55 2016
 OS/Arch:      windows/amd64

Server:
 Version:             1.14.0-dev
 API version:         1.26
 Minimum API version: 1.24
 Go version:          go1.7.3
 Git commit:          e308f6b
 Built:               Tue Nov 22 18:15:55 2016
 OS/Arch:             windows/amd64
 Experimental:        false
PS C:\WINDOWS\system32> docker pull microsoft/iis
Using default tag: latest
latest: Pulling from microsoft/iis
9c7f9c7d9bc2: Extracting [==================================================>] 3.738 GB/3.738 GB
d33fff6043a1: Download complete
b7c3bbe47489: Download complete
1028e286eae2: Download complete
28e6f26ee039: Download complete
failed to register layer: re-exec error: exit status 1: output: ProcessUtilityVMImage C:\ProgramData\Docker\windowsfilte
r\a4a27a9999f248929ff4c574049f28e01e79ac28d0586ab45fe25e2a84db5d1a\UtilityVM: The process cannot access the file because
 it is being used by another process.
```

**Comment by gickis** [comment]
Please share any information how to deal with this issue. I can't uninstall symantec by corporate policy.

**Comment by Bhargav108** [comment]
It didn't worked even after uninstalling Symantec Endpoint Protection.
Kindly share any information how to deal with this issue.

**Comment by werwolfby** [comment]
The same for me, it doesn't work

**Comment by masoud-al** [comment]
Same problem here, but Unfortunately I cannot disable or uninstall the Antivirus in the my Corporate laptop.

Containers: 0
 Running: 0
 Paused: 0
 Stopped: 0
Images: 0
Server Version: 1.13.0-rc3
Storage Driver: windowsfilter
 Windows:
Logging Driver: json-file
Plugins:
 Volume: local
 Network: l2bridge l2tunnel nat null overlay transparent
Swarm: inactive
Default Isolation: hyperv
Kernel Version: 10.0 14393 (14393.321.amd64fre.rs1_release_inmarket.161004-2338)
Operating System: Windows 10 Pro
OSType: windows
Architecture: x86_64
CPUs: 4
Total Memory: 15.78 GiB
Name: ID: 
Docker Root Dir: C:\ProgramData\docker
Debug Mode (client): false
Debug Mode (server): false
Registry: https://index.docker.io/v1/
Experimental: false
Insecure Registries:
 127.0.0.0/8
Live Restore Enabled: false

**Comment by ecornell** [comment]
Same issue here and I have Symantec SEP (12.1.7004.6500) installed.  Its a work laptop, so I can not uninstall the antivirus solution.

--

`λ docker pull microsoft/nanoserver
Using default tag: latest
latest: Pulling from microsoft/nanoserver
bce2fbc256ea: Extracting [==================================================>] 252.7 MB/252.7 MB
10bf725c5388: Download complete
failed to register layer: re-exec error: exit status 1: output: ProcessUtilityVMImage C:\ProgramData\docker\windowsfilter\721c05cbfeaa4b87eebc31cd9ea66ac1e90413c42a4806f5a6a51d29e1f5210e\UtilityVM: The process cannot access the file because it is being used by another process.

λ docker version
Client:
 Version:      1.13.0-rc3
 API version:  1.25
 Go version:   go1.7.3
 Git commit:   4d92237
 Built:        Tue Dec  6 01:15:44 2016
 OS/Arch:      windows/amd64

Server:
 Version:      1.13.0-rc3
 API version:  1.25 (minimum version 1.24)
 Go version:   go1.7.3
 Git commit:   4d92237
 Built:        Tue Dec  6 01:15:44 2016
 OS/Arch:      windows/amd64
 Experimental: false
`

**Comment by aadennis** [comment]
I am also getting the "...cannot access the file..." message. I have had the same symptom on 1.12, 1.13, and now the manual download of the latest 1.14:
![image](https://cloud.githubusercontent.com/assets/11707983/21287852/ae5a8472-c46c-11e6-967a-314df351b801.png)

![image](https://cloud.githubusercontent.com/assets/11707983/21287845/86315232-c46c-11e6-9e5f-37294598bc3e.png)

![image](https://cloud.githubusercontent.com/assets/11707983/21287829/2aca8116-c46c-11e6-897c-c34b1c8d7eec.png)

![image](https://cloud.githubusercontent.com/assets/11707983/21287898/779b553c-c46d-11e6-9b8c-8ff4d8ec4e0c.png)


The AV problem sounds a plausible explanation... but I do not want to switch mine off. Kaspersky Total Security 17, FWIW.

**Comment by viertaxa** [comment]
I am also experiencing this issue running the latest Windows 10 build with the 1.13-RC3 build of Docker. I am only running Windows Defender. I've tried adding exclusions for all of the docker related folders I could find on the system.

**Comment by tnedrow** [comment]
I also was able to install only after removing Symantec Endpoint Protection.

**Comment by shadercode** [comment]
I am also running docker version 1.14.0-dev with no success, after trying 1.13.  I have Symantec Endpoint Production as well, and I cannot disable it.  Even if I could, corporate VPN would block my access due to bad "posture".  I have tried disabling Auto-Protect, which it _appears_ to let me do.  Nothing works.  I cannot kill the symantec processes.  I have spent an entire day trying to get a container running that was provided by a vendor, and it appears I will not be able to use docker at all.

**Comment by viertaxa** [comment]
After updating to the latest beta that got pushed out today, I am receiving the following message:

failed to register layer: re-exec error: exit status 1: output: ProcessUtilityVMImage C:\ProgramData\Docker\windowsfilter\de4364db28f6e9c930804ce9896afb260cd6432616657c48a82a89469d1dca5d\UtilityVM: The parameter is incorrect.

**Comment by philb32** [comment]
I had the same problem on Windows 10 Pro, but disabling McAfee active scanning solved it. Before was getting this : 
PS C:\Users\philb\tmp> docker build -t penv/1.0 .
Sending build context to Docker daemon 3.072 kB
Step 1/3 : FROM microsoft/nanoserver
latest: Pulling from microsoft/nanoserver
bce2fbc256ea: Pull complete
10bf725c5388: Extracting [==================================================>] 59.75 MB/59.75 MB
failed to register layer: rename C:\ProgramData\Docker\image\windowsfilter\layerdb\tmp\write-set-937614473 C:\ProgramDat
a\Docker\image\windowsfilter\layerdb\sha256\b61ebdad6b389114bf5365f2398cd205e8931f4ccf857bae0a8f08c231f327b8: Access is
denied.

But after disabling active scanning, it works : 
PS C:\Users\philb\tmp> docker build -t penv/1.0 .
Sending build context to Docker daemon 3.072 kB
Step 1/3 : FROM microsoft/nanoserver
latest: Pulling from microsoft/nanoserver
bce2fbc256ea: Pull complete
10bf725c5388: Pull complete
Digest: sha256:6e2b7e3d21e385681a38a8d66a40213462762e7e14c6e97b7701cee8b21e32d6
Status: Downloaded newer image for microsoft/nanoserver:latest
 ---> 654716d8aec0
Step 2/3 : COPY print-env.ps1 c:\\print-env.ps1
 ---> ec8226baf141
Removing intermediate container 3079235028ee
Step 3/3 : CMD powershell.exe c:\print-env.ps1
 ---> Running in 81d91bc20ec9
 ---> d9378199f172
Removing intermediate container 81d91bc20ec9
Successfully built d9378199f172

This is with 1.13.0-rc4 client and server

**Comment by jabbera** [comment]
I've been upgraded to SEP 14.0.1904.0000 and all seems well.

**Comment by viertaxa** [comment]
Just applied the latest cumulative update to my machine from MS. Ran the following:


failed to register layer: re-exec error: exit status 1: output: ProcessUtilityVMImage C:\ProgramData\Docker\windowsfilter\ae244b2b50fe6431bc1abc947283994b6c7b2b00891960fcdce0e7c609e94933\UtilityVM: winapi error #3489661090

**Comment by johnvanloon** [comment]
As said in https://github.com/docker/for-win/issues/363 I'm also getting this error. Tried with Docker 1.13 and 1.14.
At the time I'm getting the error in docker I'm also getting a error on hpdskflt (see other issue for details) and I'm getting a warning in the event viewer:

An error was detected on device \Device\Harddisk1\DR2 during a paging operation.

**Comment by JanneRantala** [comment]
I'm also having the same issue. Everything was working fine few weeks ago, I didn't update Docker or anything and then suddenly last week I started getting this error. Tried updating Docker to 1.13.0-rc4 with no luck.

**Comment by fungos** [comment]
Same here. 1.13 and 1.14-dev. Disabling SEP does not work.

**Comment by johnvanloon** [comment]
Tried the workaround in https://github.com/docker/for-win/issues/292#issuecomment-266816403 . No luck, still having the same problem.

**Comment by drnybble** [comment]
Tried with the latest 1.14-dev and it did not help. Tried disabled SEP 12.x and Windows Defender, no luck. This is a major blocker! How to get some attention for this issue...

**Comment by philb32** [comment]
For what it's worth, I just upgraded from Docker 1.13.0-rc4-beta34 to
1.13.0-rc7-beta37 on Windows 10 pro, and it still seemed to be working ok.
Then I enabled McAfee realtime scanning again and tried 'docker build ...'
which failed with 'Access is denied' error, disabled McAfee realtime
scanning and 'docker build ...' worked ok. So it is definitely the McAfee
software which is preventing docker from working properly on my machine,

Cheers, Phil

**Comment by philb32** [comment]
Try using sysinternals procmon to see exactly what the problem is ... maybe
will help ...

**Comment by drnybble** [comment]
I did use ProcMon, I couldn't see any obvious errors.

**Comment by adrianvillanueva** [comment]
I'm having the same issue.

Software Versions:
Windows 10.0.14393
SEP 12.1.7004.6500
Docker 1.13.0-rc7-beta37 (9733)

Has anyone tried SEP 14?  According to [https://support.symantec.com/en_US/article.TECH235458.html](url), "Symantec Endpoint Protection 14 is fully supported on Windows 10 Anniversary Edition and Windows Server 2016".

Additionally, "To use Symantec Endpoint Protection 12.1 RU6 MP6 (and later versions of 12.1.x) with Windows Server 2016, Docker for Windows, Microsoft Hypervisor Code Integrity (HVCI) and Microsoft Device Guard must remain disabled to avoid compatibility issues"

**Comment by drnybble** [comment]
I have downloaded and installed the trial version of SEP 14 onto a clean Server 2016 VM. I am able to pull docker images without error. It does not appear that docker needs to fix anything but that you must be running the latest version of your AV software (or hound them to fix the problem).

**Comment by JanneRantala** [comment]
@drnybble I have also been able to install Docker on Windows Server 2016 but that still doesn't help me with issue on my development machine. I'd really like that Microsoft would tell what is the real issue with Docker on these pull/load/build problems and what we should do about those. 
And btw, I'm using McAfee on both Windows 10 and 2016 server (corporate policy, not that easy to remove ) and I originally had my pull issue in last autumn. It disappeared just as strangely as it appeared. And now, after several months it came back. All that time I've been using the same McAfee. So I'm a little frustrated, I'd really like to start Docker full time but at the moment it's not possible.

**Comment by drnybble** [comment]
Sorry to report that with SEP 14.0.1094.0000 while I could pull images, **docker run** hangs and never completes.

**Comment by arasheedu** [comment]
Hi All,

Having faced with same issue as many people here with SEP and DOCKER on Windows 10, I rebuilt the daemon myself (dockerd-1.14.0-dev.exe) and did some debugging on my own.

Here's what I changed in code before the call to ProcessUtilityVM in docker daemon:
Sleep for about 2 minutes (Just in case SEP is still processing the files just extracted and created )
Implemented a directory walker that traverses the folder UtilityVM\Files and try to open each file with "syscall.GENERIC_ALL" permission, if open succeeds I immediately close the file.
Keep a count of Files I am able to open in the above step and files that I am not able to open
Then proceed to the ProcessUtilityVM call ( which fails as usual saying files are being used by some other process )

What I found (for microsoft/nanoserver):
2017-01-25 10:04:59.960935 I | Opened 438 files
2017-01-25 10:04:59.960935 I | Could not open 4709 files
The files I could not open are either in "\UtilityVM\Files\EFI\Microsoft\Boot" folder tree or "UtilityVM\Files\Windows" folder tree

I had setup SEP folder exclusion for the entire "C:\ProgramData\docker\windowsfilter" folder. But results are still the same.

Can any experts here deduce anything from this such that we can have a fix at the earliest?

Thanks!

**Comment by JanneRantala** [comment]
Nice work, @arasheedu! I would certainly hope that someone from Microsoft would start taking these issues seriously and start fixing them, these are major blockers at them moment.

**Comment by johnvanloon** [comment]
We have been investigating this issue yesterday again. We already tried excluding the specified folders from Windows Defender earlier without luck.

We used a clean laptop, exactly same model, but without our policies and without joining it into our domain. On that machine everything worked as expected. We then joined this machine to our domain and then the issue also appeared on this machine.

For us this means it has something to do with our policies or domain settings and this rules out the possibility it is related to a hardware/driver issue. We are currently getting in contact with Microsoft support to get this issue fixed. To give them as much information as possible I have 2 questions for the people experiencing this same problem:
* Is your system also joined to a domain?
* Does the issue also disappear when using the same machine without joining a domain?

Thanks.

**Comment by viertaxa** [comment]
@johnvanloon 

I can second your experiences exactly, however we don't have any policies on the domain that I could find that should have an effect on containers/docker. 

Additionally, I see two errors occurring in this issue on Github, with what I believe to be two different root causes.

The first error is the "access denied" or "file is in use" error caused by AV. This seems to be mitigated by disabling or uninstalling AV.

The second, which I and, presumably, you are experiencing is the "Parameter is incorrect" or "WinAPI" error. This seems to be related to machines joined to a domain, regardless of hardware. OS level, patches, etc.

Hopefully MS can look in to this issue soon, as it is causing major hangups for our team.

Additionally, I would suggest creating a new issue for the AV related errors.

**Comment by arasheedu** [comment]
@JanneRantala - Thank you!
@johnvanloon - My system is joined to a domain, not sure of the result if I am not joined to a domain.

Thanks!

**Comment by JanneRantala** [comment]
I'm with that  WinAPI -issue and what might be because of domain. To test that, what will happen if i'll just remove my laptop from domain and add it to workgoup? I mean can I just test it like that and what will be the side effects on my laptop? Or are there any, @johnvanloon?

**Comment by sg1970** [comment]
@johnvanloon

My system is a **standalone** laptop running Win10 build 14393.693. (Tried disabling Windows Defender no luck). I do not have the option of joining a domain. Any help will be greatly appreciated. Cant believe that this issue was first reported  in Aug,!  (But pretty sure that I didn't have this issue earlier in Jan )

Docker version = 1.13.0-beta39 (9861)

-----
PS C:\> docker version
Client:
 Version:      1.13.0
 API version:  1.25
 Go version:   go1.7.3
 Git commit:   49bf474
 Built:        Tue Jan 17 21:19:34 2017
 OS/Arch:      windows/amd64

Server:
 Version:      1.13.0
 API version:  1.25 (minimum version 1.24)
 Go version:   go1.7.3
 Git commit:   49bf474
 Built:        Tue Jan 17 21:19:34 2017
 OS/Arch:      windows/amd64
 Experimental: true

PS C:\> 

**Error Details while building a simple hello world .net core application image FROM microsoft/dotnet:1.1.0-runtime-nanoserver base image ** 

-------

C:\Docker\Test>docker build -t testcore03 .
Sending build context to Docker daemon 36.57 MB
Step 1/4 : FROM microsoft/dotnet:1.1.0-runtime-nanoserver
1.1.0-runtime-nanoserver: Pulling from microsoft/dotnet
bce2fbc256ea: Already exists
3ac17e2e6106: Extracting [==================================================>] 80.62 MB/80.62 MB
6c31d184df01: Download complete
babf60756adf: Download complete
4bada20dcb10: Download complete
0cbd8997555c: Download complete
fc6441addebf: Download complete
failed to register layer: re-exec error: exit status 1: output: remove \\?\C:\ProgramData\Docker\windowsfilter\f798a1301246f4d808bcf37c9b475f0dbcb1d63cb5e26745544bc4cfd6636633\UtilityVM\Files\Windows\servicing\Packages\Microsoft-OneCore-Containers-Package~31bf3856ad364e35~amd64~~10.0.14393.0.cat: The process cannot access the file because it is being used by another process.

C:\Docker\Test>

**Comment by taylorb-microsoft** [comment]
@adrianvillanueva see previous comment from @jabbera on Dec 23 2016 

> I've been upgraded to SEP 14.0.1904.0000 and all seems well.

**Comment by drnybble** [comment]
@taylorb-microsoft Not for me -- with SEP 14.0.1904.0000 I can pull images but **docker run** hangs. Uninstall SEP 14.0.1904.0000 and it works. This is with the latest 1.14-dev docker.

**Comment by viscropst** [comment]
I have not been install the any anti-virus software,but I have the same problem.

**Comment by drnybble** [comment]
@viscropst Do you have a fully patched system? That error is not the same as others have reported.

**Comment by viscropst** [comment]
My Server 2016 Standard has the newest upgrade,and fully patched system.

**Comment by philb32** [comment]
Is Microsoft taking any notice of this?

**Comment by viscropst** [comment]
No,I am not talking about this to Microsoft.

**Comment by dermeister0** [comment]
Docker stopped to load when I moved all images to another drive.

{code}
Version: 1.13.0-rc5-beta35 (9606)
Channel: Beta
Sha1: d603c07c049bd3ef899e3ac09ead2a5db339679d
Started on: 2017/02/14 10:41:39.617
Resources: C:\Program Files\docker\Docker\Resources
OS: Windows 10 Pro
Edition: Professional
Id: 1607
Build: 14393
BuildLabName: 14393.693.amd64fre.rs1_release.161220-1747
{code}

{code}
time="2017-02-14T11:20:47.596366700+07:00" level=info msg="Loading containers: start." 
time="2017-02-14T11:20:47.607368300+07:00" level=debug msg="WindowsGraphDriver Get() id 00597a29d1b8b70e66668da190e65e3e17741bc59891be51165a8ec165fee34e mountLabel " 
time="2017-02-14T11:20:47.610870000+07:00" level=debug msg="hcsshim::ActivateLayer Flavour 1 ID 00597a29d1b8b70e66668da190e65e3e17741bc59891be51165a8ec165fee34e" 
time="2017-02-14T11:20:47.792737700+07:00" level=error msg="hcsshim::ActivateLayer failed in Win32: The process cannot access the file because it is being used by another process. (0x20) id=00597a29d1b8b70e66668da190e65e3e17741bc59891be51165a8ec165fee34e flavour=1" 
time="2017-02-14T11:20:47.792737700+07:00" level=error msg="Failed to mount container 00597a29d1b8b70e66668da190e65e3e17741bc59891be51165a8ec165fee34e: hcsshim::ActivateLayer failed in Win32: The process cannot access the file because it is being used by another process. (0x20) id=00597a29d1b8b70e66668da190e65e3e17741bc59891be51165a8ec165fee34e flavour=1" 
{code}

I used the [Handle](https://technet.microsoft.com/en-us/sysinternals/bb896655.aspx) tool to see what process locks the files, it's System.

{code}
$ handle D:\Docker\windowsfilter\00597a29d1b8b70e66668da190e65e3e17741bc59891be51165a8ec165fee34e

Nthandle v4.1 - Handle viewer
Copyright (C) 1997-2016 Mark Russinovich
Sysinternals - www.sysinternals.com

System             pid: 4      type: File          127C: D:\Docker\windowsfilter\00597a29d1b8b70e66668da190e65e3e17741bc59891be51165a8ec165fee34e\sandbox.vhdx
System             pid: 4      type: File          16E8: D:\Docker\windowsfilter\00597a29d1b8b70e66668da190e65e3e17741bc59891be51165a8ec165fee34e\sandbox.vhdx
{code}

**Comment by JanneRantala** [comment]
Is anybody paying any attention to these issues which have been open for ages? Microsoft? Somebody? Anybody? This is really unacceptable.

**Comment by PatrickLang** [comment]
Thanks for the updates on this! I'm sorry we're behind on these issues. I'd like to be able to respond to every one but unfortunately I'm only able to follow GitHub occasionally with other work on my plate :(

This issue seems to have several themes in it which could be separate issues:
- Symantec Endpoint Protection was blocking some file accesses. Based on what I've heard from Symantec this was fixed in SEP in the latest update as of 2/14/2017. 
- There were various ranges of Docker builds during v1.13 development that had problems with `docker load` that are fixed in the final v1.13 and later versions


If you're running the latest and still hitting issues, you should be able to open a ticket with Microsoft support and get problems with your deployment driven to resolution.

**Comment by JanneRantala** [comment]
Thanks for your reply. This is not even the issue I'm facing, I'm just trying to find out something or somebody who would be able to help me with this: 

https://github.com/Microsoft/Virtualization-Documentation/issues/530

As I wrote in that bug, I have McAfee installed due to a corporate policy but even uninstalling that didn't help. I had the same problem in August-September, then suddenly it started working. Now it's been two months since that same error reappeared and I'm back to square one. Now where exactly I need to open ticket about this to get some attention? Could you provide me a link, please? Thank you in advance.

**Comment by drnybble** [comment]
@PatrickLang I have just installed SEP MP1 (14.0.2332.0100) released on February 17, 2017.
Sorry to say it still doesn't work. A simple:
  docker run --rm microsoft/windowsservercore powershell
never completes. Same error I documented above about three weeks ago including a stack from windbg.

Is this something Microsoft can pursue or are you going to have individual developers trying to chase down AV vendors?

**Comment by johnvanloon** [comment]
Sorry for the long silence from my side, but we have finally been able to solve the problems on our systems. The problem was actually twofold:

First problem
Cause: 
When setting up a new base OS image, vmcompute.dll is creating and modifying Vhdx volumes to implement the union file system needed to support containers. On our system, the “Deny write access to fixed not protected by Bitlocker” GPO prevents to modify  with “Not Bitlocker encrypted” Vhdx volume.

Solution:
Disable this GPO. I can imagine this is not an option for all of you, but for us it is for now.

Second problem was solved by installing the cumulative update for January ( http://www.catalog.update.microsoft.com/Search.aspx?q=KB3216755 )

**Comment by friism** [comment]
@ebriney we should probably check for that GPO in the Docker for Windows installer.

@enderb-ms @PatrickLang can we also add a check in the Powershell installer?

**Comment by jhiller** [comment]
If you have local admin rights, running this reg script will allow you to pull images if you have the "Deny write access to fixed not protected by Bitlocker" group policy enabled.

[HKEY_LOCAL_MACHINE\SYSTEM\CurrentControlSet\Policies\Microsoft\FVE]
"FDVDenyWriteAccess"=dword:00000000
"RDVDenyWriteAccess"=dword:00000000

**Comment by StefanScherer** [comment]
@JanneRantala could you check the registry keys HKEY_LOCAL_MACHINE\SYSTEM\CurrentControlSet\Policies\Microsoft\FVE\*DVDenyWriteAccess of your both machines?

**Comment by JanneRantala** [comment]
@StefanScherer Checked from another of those machines and that value was enabled. When I disabled it, pull started working! This just awesome. Now I wonder will this setting stay disabled or will GPO again enable it?

**Comment by JanneRantala** [comment]
@StefanScherer I can confirm that this solution worked for both my machines. Thanks to @johnvanloon, @jhiller and everyone else who helped solving this!

**Comment by ebriney** [comment]
@friism : We are admin when we do the install so we can check it, can you add an issue on Pinata (PS: with the new installer we can imagine disabling it or change PVE...)
cc @simonferquel

**Comment by JanneRantala** [comment]
Just as I expected, group policy has again enabled that registry value. Is there any other way to fix this than editing registry every time, @StefanScherer @PatrickLang?

**Comment by JanneRantala** [comment]
Btw I can't understand why this GPO even does this, it says that "Deny write access to fixed not protected by Bitlocker", yes I'm using Bitlocker, but both my drives where I've tried running Docker are also Bitlocker protected. @StefanScherer @PatrickLang

**Comment by PatrickLang** [comment]
Group Policy settings are set by your domain administrator. This lets them push consistent settings down to all devices. https://technet.microsoft.com/en-us/library/ee706521(v=ws.10).aspx#BKMK_driveaccess1 

This setting indirectly affects containers because we create a VHD for each container's scratch space and format it. Since it's a new disk and not yet encrypted, then Windows will block access to that VHD based on that policy.

**Comment by JanneRantala** [comment]
That makes sense I guess. Thanks for clearing it out, @PatrickLang. Is there anything you can do about it or do I have to try to get an exception for that policy?

**Comment by PatrickLang** [comment]
I'm starting that discussion with the devs but it would take a change to Windows to work around this which means there's no quick fix.

I think you will need an exception for this.

**Comment by JanneRantala** [comment]
Ok thanks. I'll try to get that exception while waiting for Windows fix.

**Comment by JanneRantala** [comment]
Couldn't get policy exception for this. So I've done Powershell script which changes registry key before doing docker pull/build etc. This is definitely not bulletproof solution but I have to live with until proper fix is available. Is there a place where I can follow progress on this, @PatrickLang?

**Comment by PatrickLang** [comment]
@JanneRantala Can you open an issue regarding that key on https://windowsserver.uservoice.com/forums/304624-containers ? I don't really have a better public place to track it.

**Comment by drnybble** [comment]
Reiterating my question above -- SEP 14 MP1 still does not work with docker on Windows. Is this something Microsoft will pursue with the AV vendor (since it might require Microsoft developer assistance to diagnose and resolve) or will you have a developer try to open a case with Symantec.

**Comment by s2zaman** [comment]
May be this post on issue-97 solve your issue 👍

Have a look here at [#issuecomment-258271663](https://github.com/docker/labs/issues/97#issuecomment-258271663) .

**Comment by drnybble** [comment]
@PatrickLang With reference to VHD and BitLocker above -- I assume that the VHD will be encrypted after creation?

**Comment by PatrickLang** [comment]
@drnybble no, but it doesn't really need to be. The host drive should still be Bitlocker encrypted based on group policy. There isn't any support for per-container Bitlocker encryption keys

**Comment by viertaxa** [comment]
The VHD is encrypted from the start because it is being written to a bitlocker encrypted partition already.

**Comment by nullberri** [comment]
I too am having this exact issue, however I do not have...

- 3rd party anti-virus,
- bit locker is not installed
- disabled windows search and windows defender.  

`
Windows build 10.0.14393 · Version 1607 and all updates possible have been applied. 
Docker versions:
Client:
 Version:      17.03.1-ce
 API version:  1.27
 Go version:   go1.7.5
 Git commit:   c6d412e
 Built:        Tue Mar 28 00:40:02 2017
 OS/Arch:      windows/amd64

Server:
 Version:      17.03.1-ce
 API version:  1.27 (minimum version 1.24)
 Go version:   go1.7.5
 Git commit:   c6d412e
 Built:        Tue Mar 28 00:40:02 2017
 OS/Arch:      windows/amd64
 Experimental: true`

**Comment by mylesbkeating1993** [comment]
@nullberri  there are a lot of issues mentioned in this thread haha :)

Could you please tell us which error message you're getting and what command you're trying to run?  

Is it "The process cannot access the file because it is being used by another process."?

**Comment by nullberri** [comment]
@mylesbkeating1993 

Yes its the same error as the original poster.
Running docker pull Microsoft/windowsservercore or docker load -i Microsoft.tar (which i got from a different machine that's working)

**Comment by mylesbkeating1993** [comment]
@nullberri 

Do you have any antivirus software installed?

See [thread mentioned above](https://social.msdn.microsoft.com/Forums/en-US/13523ccf-baf4-49fc-9e73-1df07e512304/reexec-error-when-trying-to-load-nanoserver-image?forum=windowscontainers)

**Comment by nullberri** [comment]
@mylesbkeating1993
As stated I do not have any anti-virus installed. 

I initially tried with just disabling windows defender real-time protection and adding exemptions for all of the docker folders (program files and program data locations). When that did not resolve it so I also disabled windows search in case it was trying to index the files while running. 

I have also tried installing docker ee on a fresh install of windows server 2016 with nothing installed except windows updates, the container feature and docker, same results. 

At this point I'm more inclined to believe its hardware related. As the issue does not happen on two other machines I have.

For what its worth the machine having the issue was built with:
Intel i5-6500
gigabyte H170N0-WIFI-CF (Bios F4c)
Crucial Technology 16gb of DDR4-2132 part CT8G4DFD8213.C16FA
no discreet gpu.

**Comment by mylesbkeating1993** [comment]
@nullberri Man this is frustrating.  I have no idea.  

I'll circle up with everyone and see if anyone has seen something like this.

Are there any other possible differences between this machine that doesn't work and the ones that do work besides hardware (e.g. network)?

**Comment by nullberri** [comment]
@mylesbkeating1993 

As it turns out I had one difference, [Stable bit drive pool](https://stablebit.com/DrivePool). When I disabled the service, docker pull/load worked. Man that's frustrating. I have no idea why it would be reading from the C drive when that drive is not apart of the drive pool. 

I'll open a support ticket with them.

**Comment by mylesbkeating1993** [comment]
Hey @nullberri excellent!!  Glad you figured it out.  

I'm not exactly sure what you're saying is the issue, but it sounds like something about Stable bit was causing the Docker daemon to read from the C drive?  Is that right?

What are you trying to accomplish with Stable Drive?  Keeping all your files across multiple machines organized in a single place?

Please tag me on the support ticket - would love to work with them to understand and resolve this :)

For future issues, @friism turned me on to some tools that might be able to help you out with identifying exactly which process is messing things up:

https://technet.microsoft.com/en-us/sysinternals/sysmon
https://technet.microsoft.com/en-us/sysinternals/bb896645

Let us know if you have any further issues or questions!

**Comment by nullberri** [comment]
@mylesbkeating1993

> I'm not exactly sure what you're saying is the issue, but it sounds like something about Stable bit was causing the Docker daemon to read from the C drive? Is that right?

The other way around. Something drivepool is doing is blocking docker from running ProcessUtilityVMImage  on the utlityvm folder.

 > What are you trying to accomplish with Stable Drive? Keeping all your files across multiple machines organized in a single place?

Correct its a form of software raid that has a number of benefits over traditional raid. 

[Post on stablebit community forms](http://community.covecube.com/index.php?/topic/2780-drive-pool-interfering-with-docker-for-windows/)

**Comment by mylesbkeating1993** [comment]
@nullberri 

As per the [community forum post](http://community.covecube.com/index.php?/topic/2780-drive-pool-interfering-with-docker-for-windows/) it appears the magic order of:

1. Remove SSD in question from drive pool
2. Have docker on the SSD
3. Re-add the SSD to the bitpool

solves your issue?

**Comment by dermeister0** [comment]
> Docker version 17.05.0-ce-rc1, build 2878a85

Resolved the issue when added `C:\ProgramData\Docker` to Windows Defender exclusions and restarted PC.

**Comment by artisticcheese** [comment]
I'm getting "The parameter is incorrect" error at the very end of extraction of base layer. Defender has C:\ and D:\ excluded. No Bitlocker or any sort of encryption. Running latest Windows 10.


  - Diagnostic ID from "Diagnose & Feedback" in the menu.
2A2F175C-7BAC-4994-AE7C-1D22FD255891/2017-05-11_13-53-30
  - a reproducible case if this is a bug, Dockerfiles FTW
docker pull microsoft/nanoserver
  - page URL if this is a docs issue or the name of a man page
  - host distribution and version (Windows version, build number, etc)
Windows 10 1607, build 14393.1066
### Steps to reproduce the behavior
Docker version
```
Client:
 Version:      17.05.0-ce
 API version:  1.29
 Go version:   go1.7.5
 Git commit:   89658be
 Built:        Thu May  4 21:43:09 2017
 OS/Arch:      windows/amd64

Server:
 Version:      17.05.0-ce
 API version:  1.29 (minimum version 1.24)
 Go version:   go1.7.5
 Git commit:   89658be
 Built:        Thu May  4 21:43:09 2017
 OS/Arch:      windows/amd64
 Experimental: true
```

Repro

```
PS C:\gd\baseimage> docker pull microsoft/nanoserver

Using default tag: latest
latest: Pulling from microsoft/nanoserver
bce2fbc256ea: Extracting [==================================================>]  252.7MB/252.7MB
4a8c367fd46d: Download complete
failed to register layer: re-exec error: exit status 1: output: ProcessBaseLayer D:\images\windowsfilter\f688510f3cbacba828f7395e0ab9a270f688f65d3c58abb47f8883f3db81
83f8: The parameter is incorrect.
```

**Comment by gaya3chandran1** [comment]
@drnybble  I am also facing the same error

Error response from daemon: encountered an error during Start: failure in a Windows system call: This operation returned because the timeout period expired. (0x5b4).

Any solution to fix it?

**Comment by kiyakkoray** [comment]
I have the same problem as you do. Disabling anti virus software does not works. I also tried to add exception rules to av and it didn't work too. Only removing av works but its against our corporation policy.  This topic seems to open nearly for a year and still no proper solution. Any ideas? 

`C:\Users\koray>docker pull microsoft/nanoserver
Using default tag: latest
latest: Pulling from microsoft/nanoserver
bce2fbc256ea: Extracting [==================================================>] 252.7 MB/252.7 MB
4a8c367fd46d: Download complete
failed to register layer: re-exec error: exit status 1: output: ProcessUtilityVMImage C:\ProgramData\Docker\windowsfilter\32aaac762f3a60ee92ce65f24de79e8c76e48ac2e92da439a086e8d2ef14c17f\UtilityVM: The process cannot access the file because it is being used by another process.`

**Comment by artisticcheese** [comment]
@kiyakkoray Check if you have any AV running.

**Comment by kiyakkoray** [comment]
@artisticcheese Thanks for a quick reply:) Got kaspersky endpoint security.

**Comment by artisticcheese** [comment]
@kiyakkoray I would assume appropriate venue will be opening a case with Kaspersky, showing them proof that it's caused by AV and forcing them to fix this. Windows containers are still pretty knew and I already saw some puzzled looks when opening case with Mcafee so it will be uphill battle.

**Comment by gaya3chandran1** [comment]
Hi

Thanks for the comment. Yes it is AV issue here too and its against policy
to remove it.Another  work around?

Regards
Gayathri

**Comment by artisticcheese** [comment]
There is no fix Microsoft or docker can provide for third party AV. If AV is causing issue why complaint here? It shall be escalated directly to AV vendor, by doing this you will help other users as well since by the time they start using windows containers AV will be fixed. Based on thread so far looks like only Defender properly handles containers.

**Comment by artisticcheese** [comment]
Here is official Microsoft guidance for AV manufactureres how to deal with Windows Containers. So when creating cases with AV vendors reference this article. Symantec and Defender are the only ones which seems to be on bandwagon right now.
https://docs.microsoft.com/en-us/windows-hardware/drivers/ifs/anti-virus-optimization-for-windows-containers

**Comment by drnybble** [comment]
Seems that the latest version of Symantec is still not working: https://www.symantec.com/connect/forums/sep-14-stops-docker-windows-working-0

"Proactive Threat Protection" seems to be the culprit. On my installation of 14 MP1 I do not have that component which is why I guess it is working.

**Comment by eduard-t** [comment]
I'm getting this error when I'm using windows containers but not when using linux containers.

`failed to register layer: re-exec error: exit status 1: output: ProcessUtilityVMImage C:\ProgramData\Docker\windowsfilter\bfb413e53c1fbb3b69238d94146efefe04bae47de114c16ae12e8a282e36fffe\UtilityVM: The process cannot access the file because it is being used by another process.`

I'm using Kaspersky Endpoint Security 10. Tried disabling it, excluding multiple paths but still no luck.

**Comment by eduard-t** [comment]
just an update...

I've uninstalled Kaspersky Endpoint Security 10 and `docker pull microsoft/nanoserver` works again

**Comment by DavidHaitch** [comment]
I believe I've discovered a cause besides anti-virus problems. In my case, it was one particular file that was always locked:
`PS Q:\> docker pull microsoft/windowsservercore
Using default tag: latest
latest: Pulling from microsoft/windowsservercore
3889bb8d808b: Already exists
da87b55a9b63: Extracting [==================================================>] 1.266GB/1.266GB
failed to register layer: re-exec error: exit status 1: output: remove \\?\D:\DockerData\windowsfilter\9c420937b1d524adbaefdbf7d1b9f6af927baff9680ea3587861656b70952645\UtilityVM\Files\Windows\servicing\Packages\
Microsoft-NanoServer-Win32Compat-Package~31bf3856ad364e35~amd64~en-US~10.0.14393.0.cat: The process cannot access the file because it is being used by another process.
`

It was always that one NanoServer-Win32Compat package. Disabling antivirus had no effect. What did fix this was going to C:\Windows\Temp, and deleting every folder that began with "hcs".

It looks like these folders are created by the Host Compute Service during the image creation process. They're supposed to be cleaned up at the end of the build, but mine weren't. Apparently they were holding some kind of lock, and that lock was interfering with both the extraction process, and the Temp cleanup process.

I hope this info helps anyone else who has this issue!

**Comment by JanneRantala** [comment]
Hi @DavidHaitch! I've noticed, that the same file as you mentioned is being used by another process in my environment as well. Sometimes I've managed to get over it by killing vmcompute -process but yesterday things got so stuck up that I had to zap my whole docker folder. Now everything seems to work alright again, but the next time I'll get that error I'll try your fix. Thanks!

**Comment by DavidHaitch** [comment]
@JanneRantala If you do see this again, try pulling microsoft/windowsservercore-insider first. That one worked for me when the regular windowsservercore did not.

**Comment by JanneRantala** [comment]
@DavidHaitch I'm not using insider version of Windows 10.

**Comment by DavidHaitch** [comment]
@JanneRantala  Neither am I. The windowsservercore-insider container worked nevertheless. Such is the magic of containerization.

**Comment by PhilStanton** [comment]
Same problem. nothing fixing this yet for me.

**Comment by PhilStanton** [comment]
Tried adding folders to exclusions in Kaspersky, still not helping. Having to uninstall anti-virus to get Docker for Windows to work, if that does fix it, is a bit much.

**Comment by brunofvalli** [comment]
I have found if I stop and remove the running containers on my docker it seems to solve the issue. So I guess I need a build box that I will never run a container on it.

**Comment by nanoant** [comment]
@aphophas I am curious about this SEP exclusion for `C:\ProgramData\docker` that worked for you, but not for me. Which version of SEP you are using (12 or 14)?

**Comment by wadinj** [comment]
Nothing interesting works for me here. For sure, I didn't try to uninstall my SEP (corporate policy as everyone). But, disable SEP and BitLocker change nothing, always stuck to : 
`C:\Program Files\Docker\Docker\Resources\bin\docker.exe: failed to register layer: re-exec error: exit status 1: output:
 ProcessUtilityVMImage C:\ProgramData\Docker\windowsfilter\cc9a78038e66f96bd8d0851d02a5a825d809497f1980b691d10748fe36efa
d1e\UtilityVM: The process cannot access the file because it is being used by another process.`

I guess one good solution will be to add a timeout option in docker to wait for availability of the internal file ?

**Comment by nanoant** [comment]
FYI (@wadinj et al.) this is not Docker fault, it is `ProcessUtilityImage` in `vmcompute.dll` aka Windows Host Compute API (part of Windows OS) used by Docker via https://github.com/Microsoft/hcsshim, that is faulting with `ERROR_SHARING_VIOLATION` exception that is simply brought back and reported by Docker.

There is nothing Docker can do about it. Since `ProcessUtilityImage` doing kind lot of things creating virtual HD image etc. and it fails at some stage.

I wrote myself simple Go app that is just calling `ProcessUtilityImage` on NanoServer Docker extracted image, and managed to replicate the behavior. However I found no workaround, neither SEP v12 exceptions helped. Something there (probably SEP or other AV) definitely peeked (opened the handle) to `vhdx` file generated by `ProcessUtilityImage` and when when it wanted to lock the file for further processing sharing violation is emitted.

So is there is anything that can be fixed (or improved) it has to be done in `vmcompute.dll` owned by Microsoft, which I believe won't happen anytime soon.

I am currently waiting for SEP v14 upgrade, once I upgrade I'll let you know if this fixes the problem on my side.

**Comment by wadinj** [comment]
@nanoant Great analyze ! I didn't spend more time on and finally switch to Linux container with toolbox.
For sure, if you figure out a solution, I'm interested. 

Thanks !

**Comment by judavi** [comment]
I face the same issue on WS2016, and was basically because I was trying to "store" the images in the D drive. So after remove the setting  "data-root": "D:\\dockerimages" from the config now is working well

**Comment by wadinj** [comment]
Finally this error went away when I downloaded the edge version of Docker to replace the stable one.

**Comment by fakoua** [comment]
Have you tried to disable the compression on the folder: C:\ProgramData\Docker?

Just try that, Right click ont he folder docker -> properties -> advanced -> uncheck "Compress contents to save disk space"

Do it while docker is off.

Thanks

**Comment by nanoant** [comment]
Just to let you know, I am now at SEP 14 RU1 MP1 and everything is working as expected finally, so it is definitely SEP 12 issue. @mduft @aphophas @ecornell @wadinj I recommend upgrading to SEP 14 then.

**Comment by JoshSchreuder** [comment]
I tried in a fresh Windows 10 Enterprise N 1709 VM and installed McAfee Endpoint Protection 10.5.3.3178.

Out of the box the `docker pull` command worked fine.

Running 10.5.1 elsewhere and it has this issue, not 100% if it's McAfee or some other environmental issue.

**Comment by gitfool** [comment]
I have the same issue on Windows Server 1709 using current latest everything:
```
C:\>ver
Microsoft Windows [Version 10.0.16299.371]

C:\>docker version
Client:
 Version:      17.10.0-ee-preview-3
 API version:  1.33
 Go version:   go1.8.4
 Git commit:   1649af8
 Built:        Fri Oct  6 17:52:28 2017
 OS/Arch:      windows/amd64

Server:
 Version:      17.10.0-ee-preview-3
 API version:  1.34 (minimum version 1.24)
 Go version:   go1.8.4
 Git commit:   b8571fd
 Built:        Fri Oct  6 18:01:48 2017
 OS/Arch:      windows/amd64
 Experimental: true

C:\>docker pull microsoft/dotnet-framework-build:4.7.1-windowsservercore-1709
...
failed to register layer: re-exec error: exit status 1: output: remove \\?\C:\ProgramData\docker\windowsfilter\31f2dab02e9b94b1bbc14ceb542522e493057c128c02d571936955b71db68d04\UtilityVM\Files\Windows\WinSxS\amd64_microsoft-windows-workstationservice_31bf3856ad364e35_10.0.16299.15_none_ef2643e047f6349e\wkssvc.dll: Access is denied.
```
I only have the built-in Windows Defender feature installed, so I tried excluding the Docker folder:
```
Set-MpPreference -ExclusionPath "C:\ProgramData\docker"
```
Then I tried disabling Windows Defender realtime monitoring:
```
Set-MpPreference -DisableRealtimeMonitoring $true
```
Then I tried removing the Windows Defender feature:
```
Uninstall-WindowsFeature -Name "Windows-Defender"
Restart-Computer
```
Finally, I tried @brunofvalli 's suggestion and **I stopped all other running containers before pulling the container and it worked. Eureka!**

**Comment by janainaemilia** [comment]
@RustyCZ  You rock! I had the same problem when execute `docker pull microsoft/mssql-server-windows-express`, and when I stopped the others containers and execute pull again, it works! Thank you 💯

**Comment by filou77** [comment]
I have got the same error.
Then I realized that I tried to use a version of windowsservercore that was unsupported by my host OS according to:
https://docs.microsoft.com/en-us/virtualization/windowscontainers/deploy-containers/version-compatibility

**Comment by eriksegecs** [comment]
i got the same problem here, running windows server 2019 with SEP 14.3 MP1, had to fully uninstall SEP for docker to run normaly

**Comment by robinharwood** [comment]
We hugely appreciate your contribution, feedback, and support to our technical documentation.

We’ve phased out GitHub Issues on this repository as the feedback mechanism for content and have replaced it with a new feedback system. For more information see, [Content User Feedback](https://aka.ms/ContentUserFeedback). As part of this rollout, we are migrating your issue where our team will use your comments to improve the documentation experience. Unfortunately, this does mean you won't be able to access this issue after the migration is complete.

All issues will be migrated by Thursday 29th August 2024.

Once again, thank you for your valuable feedback, we hope you’ll continue to contribute using our new feedback system.
